### DISCUSSION TITLE
3D model support

### DISCUSSION THREAD

**Discussion by ChiMaoShuPhy** [marp-team/marp#323] [Ideas] [answered]
Is it possible to load and render 3D model in marp exported html slide deck? This would be extremely helpful in teaching presentation of math, physics etc.

**Comment by yhatt** [comment, marked as answer]
Marp would never support due to very limited use cases. Try to use 3rd party library, with enabling HTML in Markdown (`--html` in Marp CLI).

- [model-viewer](https://modelviewer.dev/)
- [Three.js](https://threejs.org/)

> Common pitfall: If your slide has located in the local file system (`file://`), the browser would not be able to load a model file through `fetch()`. Upload your slide and resources to the remote server, or spawn a local server with [Marp CLI's server mode `--server`](https://github.com/marp-team/marp-cli#server-mode---server---s).
